File: Sources/SPFNHybrid/SPFNHybrid.swift
// SPFN Mobile — Step 1 scaffold. No behaviour, no release promise.

import SPFNAuth
import SPFNCore

/// WebView adapter surface for hybrid apps.
///
/// The compatibility gate (topology artifact §6) requires proving that this module
/// exposes no browser-redirect auth route, no credential injection into web
/// content, and no generic JavaScript bridge. Step 1 satisfies that trivially by
/// exposing no bridge at all.
public enum SPFNHybrid
{
    /// The bridge message names this SDK will accept. Empty by design in Step 1;
    /// a generic passthrough bridge is prohibited, so this list stays explicit.
    public static let allowedBridgeMessageNames: [String] = []

    /// Auth profiles the hybrid surface may participate in. Mirrors the core
    /// allowlist; the hybrid layer never widens it.
    public static let allowedAuthProfiles: [SPFNAuthProfile] = SPFNAuthPolicy.allowedProfiles

    /// Deliberately unimplemented. No credential ever crosses into web content.
    public static func attachBridge(named _: String) throws -> Never
    {
        throw SPFNScaffoldError.notImplementedInScaffold(
            symbol: "SPFNHybrid.attachBridge(named:)",
            plannedStep: "Step 3+ — hybrid adapter expansion"
        )
    }
}

File: android/spfn-hybrid/src/main/kotlin/xyz/superfunction/spfn/hybrid/SpfnHybrid.kt
// SPFN Mobile — Step 1 scaffold. No behaviour, no release promise.

package xyz.superfunction.spfn.hybrid

import xyz.superfunction.spfn.auth.SpfnAuthPolicy
import xyz.superfunction.spfn.auth.SpfnAuthProfile
import xyz.superfunction.spfn.core.SpfnNotImplementedInScaffoldException

/**
 * Android counterpart of the Swift `SPFNHybrid`.
 *
 * The compatibility gate requires proving this module exposes no browser-redirect
 * auth route, no credential injection into web content, and no generic JavaScript
 * bridge. Step 1 satisfies that by exposing no bridge at all.
 */
object SpfnHybrid
{
    val ALLOWED_BRIDGE_MESSAGE_NAMES: List<String> = emptyList()

    val ALLOWED_AUTH_PROFILES: List<SpfnAuthProfile> = SpfnAuthPolicy.ALLOWED_PROFILES

    fun attachBridge(name: String): Nothing
    {
        throw SpfnNotImplementedInScaffoldException(
            symbol = "SpfnHybrid.attachBridge(name=$name)",
            plannedStep = "Step 3+ - hybrid adapter expansion"
        )
    }
}
